### PR TITLE
Fix: convert AppDelegate

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Convert `AppDelegate` to Swift

### CODE REVIEW

File: Wire-iOS/Sources/AppDelegate+URL.swift [hunk 1]
```diff
@@ -19,11 +19,11 @@
...
 
 extension AppDelegate {
-    @objc
-    func open(url: URL, options: [UIApplication.OpenURLOptionsKey : AnyObject]) -> Bool {
+
+    func open(url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
         guard let urlHandler = sessionManager?.urlHandler else { return false }
 
-        return urlHandler.openURL(url, options: options)
+        return urlHandler.openURL(url, options: options as! [UIApplication.OpenURLOptionsKey : AnyObject])
```
Comment (typfel) [inline Wire-iOS/Sources/AppDelegate+URL.swift]: Do the force unwrapping in the guard?

File: Wire-iOS/Sources/AppDelegate.swift [hunk 2]
```diff
@@ -18,9 +18,228 @@
...
+    func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable : Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        zmLog.info("application:didReceiveRemoteNotification:fetchCompletionHandler: notification: \(userInfo)")
+        
+        launchType = (application.applicationState == .inactive || application.applicationState == .background) ? .push : .direct
+    }
+    
+    func application(_ application: UIApplication, performFetchWithCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+        zmLog.info("application:performFetchWithCompletionHandler:")
+        
+        rootViewController?.performWhenAuthenticated(){
```
Comment (typfel) [inline Wire-iOS/Sources/AppDelegate.swift]:
```suggestion
        rootViewController?.performWhenAuthenticated() {
```

File: Wire-iOS/Sources/AppDelegate.swift [hunk 3]
```diff
@@ -18,9 +18,228 @@
...
+            ZMUserSession.shared()?.application(application, performFetchWithCompletionHandler: completionHandler)
+        }
+    }
+    
+    func application(_ application: UIApplication, handleEventsForBackgroundURLSession identifier: String, completionHandler: @escaping () -> Void) {
+        zmLog.info("application:handleEventsForBackgroundURLSession:completionHandler: session identifier: \(identifier)")
+        
+        rootViewController?.performWhenAuthenticated(){
```
Comment (typfel) [inline Wire-iOS/Sources/AppDelegate.swift]:
```suggestion
        rootViewController?.performWhenAuthenticated() {
```

File: Wire-iOS/Sources/AppDelegate+URL.swift (hunk 1 above)
Comment (typfel) [reply Wire-iOS/Sources/AppDelegate+URL.swift]: But I guess the real fix is to change the urlHandler API in sync engine to match the type.

Comment (billypchan) [reply Wire-iOS/Sources/AppDelegate+URL.swift]: removed the !

Comment (typfel) [reply Wire-iOS/Sources/AppDelegate+URL.swift]: How can that be allowed? `Any` is more generic than `AnyObject`

Comment (billypchan) [reply Wire-iOS/Sources/AppDelegate+URL.swift]: Correct, but when I try to cast it the compiler say:
```
Conditional cast from '[UIApplication.OpenURLOptionsKey : Any]' to '[UIApplication.OpenURLOptionsKey : AnyObject]' always succeeds
```

**Pull request merged**